### ISSUE TITLE
Different line styles as example in the Gallery or Cookbook

### ISSUE THREAD

**Issue by KristofKoch** [GenericMappingTools/gmt#3320]
**Description of the desired feature**

As the possibilities for styling lines has greatly increased over the last couple of months (thank you, Gentlemen) it might be beneficial to document and showcase what can be achieved with the current feature set. 

I propose to either expand the [Cookbook chapter 3.13 _Specifying pen attributes_](https://docs.generic-mapping-tools.org/latest/cookbook/features.html#specifying-pen-attributes) which is a bit sparse on the possibilities or add another example in the Gallery. 

Your thoughts about this?

**Are you willing to help implement and maintain this feature?** Yes

**Comment by PaulWessel** [comment]
Hi @KristofKoch, yes of course that would be great.  The way things are implemented is that each illustration in the documentation is a script that lives in doc/scripts and it automatically becomes part of the test suite as well.  So apart from text in the relevant cookbook section, there needs to be a script and an original PostScript plot (created with PS_MEDIA = letter) in doc/script.  Then see other scripts and how they are addressed in the docs.

**Comment by KristofKoch** [comment]
Hi @PaulWessel thank you for the primer on how to edit the cookbook. 

I currently think of showing the fancy stuff in the cookbook but explaining it might take up a lot of space. I'm undecided how to get around the problem as at least for me the main reference for experimenting with new things is the cookbook. 

Are the examples considered documentation which I can reference from the cookbook? Basically show the fancy stuff in the cookbook and direct the user to example fifty something if he wants to know how it was done?

**Comment by PaulWessel** [comment]
yes, the examples are part of the documentation and can be referenced from anywhere I believe; @seisman knows how for sure.

**Comment by seisman** [comment]
Yes, examples can be referenced anywhere in the cookbook.

Currently, the "Examples Gallery" lists 52 fancy, complicated examples. It's good to show how powerful GMT is, but users may find it hard to learn these examples because each example may use 5 or more GMT modules. 

Ideally, we'd like to have a new gallery of simple examples. Each example uses 1 or 2 commands, and plot a simple basemap, a time axis or different styling lines as @KristofKoch proposed, just like the matplotlib Gallery (https://matplotlib.org/3.2.1/gallery/index.html).

Actually, GMT already has some simple and useful examples, but they're all hidden in the long cookbook.

**Comment by joa-quim** [comment]
We can this to the Thursday's meeting agenda. I would rather start another document with simpler examples than to continue expanding the Gallery.

**Comment by PaulWessel** [comment]
Yes, I agree.  We truly need to break up the docs and present more galleries like matplotlib with simple examples.  We need a path to get there since we all agree on the goal.

**Comment by KristofKoch** [comment]
Ok, I put this on the back burner until after the meeting.

**Comment by Esteban82** [comment]
Hello to everyone. As I said in the Forum, I can contribute with examples like matplotlib. I'll be tomorrow at the meeting.

**Comment by meghanrjones** [comment]
@KristofKoch, do you mind if I transfer this issue over to https://github.com/GenericMappingTools/gmt-examples? I think that it would be great to add a gallery with different line styles to the new [gmt examples site](https://www.generic-mapping-tools.org/gmt-examples/).
